### PR TITLE
chore: add message field to native transfer entity

### PR DESCRIPTION
**Pull request by bryanchriswhite**
### Changes
- replace gql schema type with vanilla TypeScript type (not necessary to be a gql schema type)
- refactor native transfer gql queries
- add message field to the native transfer entity

### CODE REVIEW

**Review by amit-solanki — APPROVED** [review]

File: src/mappings/types/messages.ts [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import {Coin} from "./common";
 
+export interface NativeTransferMsg {
+  toAddress: string;
+  fromAddress: string;
+  amount: Coin[];
```
Comment (daeMOn63) [inline src/mappings/types/messages.ts:6]: random dumb question with zero knowledge on what's going on: what was the `@index` annotation on the previous type, and isn't it needed anymore?

**Review by daeMOn63 — APPROVED** [review]

File: src/mappings/types/messages.ts (hunk 1 above)
Comment (bryanchriswhite) [reply src/mappings/types/messages.ts:6]: Great question!

[`@index`](https://academy.subquery.network/build/graphql.html#supported-scalars-and-types) is an annotation in the graphql schema context. It effectively tells the DB to index that corresponding column. The type that was removed from the schema was not an `@entity` and therefore represented a type but not a DB table. The code generation that the schema is an input to produces a corresponding TS type but as it's not its own entity and also not used by any other in the schema, it's more appropriate for us to just write our own TS type (as we have with others).

TLDR; I don't think the `@index` annotation had any effect given how it was applied and how we are (and are not) using the resulting TS type.
